ddtrace/pin.py

```diff
@@ -147,15 +147,10 @@ def override(
             return
 
         pin = cls.get_from(obj)
-        if not pin:
-            pin = Pin(service)
-
-        pin.clone(
-            service=service,
-            app=app,
-            tags=tags,
-            tracer=tracer,
-        ).onto(obj)
+        if pin is None:
```

Comment:
Can we do this change separately (and add a test case since it actually fixes the app, tags and tracer being set properly when there is no pin on the object)?

Reply:
I'd have thought this is equivalent to what was there previously, just one less object created (no call to `clone`) if the pin didn't exist 🤔

Reply:
oh! for some reason i didn't see that `clone` was running in both codepaths before which was the point of the refactor... 😅